Lib/test/test_enum.py

```diff
@@ -3431,17 +3431,11 @@ def test_no_members(self):
             Flag(7)
 
     def test_empty_names(self):
-        for nothing, e_type in (
-                ('', None),
-                ('', int),
-                ([], None),
-                ([], int),
-                ({}, None),
-                ({}, int),
-            ):
-            empty_enum = Enum('empty_enum', nothing, type=e_type)
-            self.assertEqual(len(empty_enum), 0)
-            self.assertRaises(TypeError, 'has no members', empty_enum, 0)
+        for nothing in '', [], {}:
```

Comment:
It works also for 0.

If use
```python
                names=None if names is _not_given else names,
```
instead of
```python
                names=names or None,
```
0 will raise expected TypeError: 'int' object is not iterable.

Reply:
Both the original code and your suggestion result in

```python
TypeError: type.__new__() argument 1 must be str, not int
```

Reply:
I get:
```
TypeError: 'int' object is not iterable
```

Reply:
I finally understood what you were saying.  Changes made.